GVFS/GVFS.Virtualization/Projection/GitIndexProjection.cs

```diff
@@ -529,12 +532,25 @@ public FileSystemTaskResult AddMissingModifiedFiles()
             {
                 if (this.modifiedFilesInvalid)
                 {
-                    FileSystemTaskResult result = this.indexParser.AddMissingModifiedFiles(this.context.Tracer, this.indexFileStream);
+                    Dictionary<string, PlaceholderListDatabase.PlaceholderData> placeholderDataByPath;
+                    List<PlaceholderListDatabase.PlaceholderData> placeholdersToKeep;
+                    this.placeholderList.GetAllEntries(out placeholderDataByPath, folderPlaceholders: out placeholdersToKeep);
+
+                    FileSystemTaskResult result = this.indexParser.AddMissingModifiedFiles(this.context.Tracer, this.indexFileStream, placeholderDataByPath, placeholdersToKeep);
                     if (result == FileSystemTaskResult.Success)
                     {
                         this.modifiedFilesInvalid = false;
                     }
 
+                    // Any paths that are not in the the index must be added to ModifiedPaths.dat so that git
+                    // reports them as untracked
+                    foreach (string path in placeholderDataByPath.Keys)

```

Comment:
I would have thought `placeholdersToKeep` are the ones we need to add. Again, not really following the purpose of the two data structures.

Reply:
`placeholdersToKeep` are the entries that we should keep in the placeholder list.  At this point in the method any entries in `placeholderDataByPath` were not found in the index and need to be added to ModifedPaths.dat